tests/basic/sharedv4_svc_test.go

```diff
@@ -749,6 +695,157 @@ var _ = Describe("{Shared4SvcUnexportExport}", func() {
 	})
 })
 
+// Stop PX on the client node and wait for the server to remove the export for client.
+// Then, do a failover and restart PX on the client node.
+var _ = Describe("{Shared4SvcClientOfflineTooLong}", func() {
+	var testrailID = 54778
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/54778
+	var runID int
+	var contexts, testSv4Contexts []*scheduler.Context
+	var workers []node.Node
+
+	JustBeforeEach(func() {
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+
+	It("has to stop PX on the client node long enough for server to remove the export then bring the node back", func() {
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("client-offline-%d", i))...)
+		}
+		testSv4Contexts = getTestSv4Contexts(contexts)
+		if len(testSv4Contexts) == 0 {
+			Skip("No test-sv4-svc apps were found")
+		}
+		workers = node.GetWorkerNodes()
+
+		Step("scale the test-sv4-svc apps so that one pod runs on each worker node", func() {
+			scaleApps(testSv4Contexts, len(workers))
+		})
+		ValidateApplications(contexts)
+
+		for _, ctx := range testSv4Contexts {
+			var vols []*volume.Volume
+			var apiVol *api.Volume
+			var err error
+			var replicaNodeIDs map[string]bool
+			var attachedNode, clientNode *node.Node
+			var failover bool
+
+			//set HA level to 2 to predict which node we fail over to (needed to avoid kvdb loss)
+			Step(fmt.Sprintf("set HA level to 2 for app %s's volume", ctx.App.Key), func() {
+				vols, err = Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				setHALevel(vols[0], 2)
+				// ValidateContext() will fail with error "volume has invalid repl value. Expected:3 Actual:2"
+				// without the line below.
+				ctx.SkipVolumeValidation = true
+				replicaNodeIDs = getReplicaNodeIDs(vols[0])
+				Expect(len(replicaNodeIDs)).To(Equal(2))
+			})
+
+			Step(fmt.Sprintf("stop PX on a client node for app %s and wait for export gone", ctx.App.Key), func() {
+				// We need at least 5 nodes to do a failover after stopping PX on the client node.
+				// Otherwise, PX will lose quorum. (This assumes that there are 3 internal kvdb nodes.)
+				failover = len(workers) >= 5
+
+				apiVol, err = Inst().V.InspectVolume(vols[0].ID)
+				Expect(err).NotTo(HaveOccurred())
+
+				attachedNode, err = Inst().V.GetNodeForVolume(vols[0], defaultCommandTimeout, defaultCommandRetry)
+				Expect(err).NotTo(HaveOccurred())
+				logrus.Infof("volume %v (%v) is attached to node %v", vols[0].ID, apiVol.Id, attachedNode.Name)
+
+				// Choose a client node to stop PX on.
```

Comment:
is there a criteria for choosing the client node to stop? 
From reading the code. Seems like there are two types of nodes we don't stop on. When:
1. the node is a replica node.
2. the node is a metadata node (and when failover is true)

I'm just hoping there's a one-line description on what client node we are looking for. For example, "client node that's not a replica node and is not metadata node".

Reply:
added more descriptive comment